packages/node_modules/@webex/plugin-meetings/src/statsAnalyzer/index.js

```diff
@@ -0,0 +1,344 @@
+/* eslint-disable require-jsdoc */
+
+import EventsScope from '../common/events/events-scope';
+import {EVENT_TRIGGERS} from '../constants';
+
+import defaultStats from './global';
+
+export default class StatsAnalyzer extends EventsScope {
+  constructor(config) {
+    super();
+    this.statsResults = defaultStats;
+    this.config = config;
+    this.packetLossArraySender = {audio: [], video: [], share: []};
+    this.packetLossArrayReceiver = {audio: [], video: [], share: []};
+  }
+
+  parseResults = (result, type) => {
+    if (!result) {
+      return;
+    }
+    this.outboundrtp(result, type);
+    this.inboundrtp(result, type);
+    this.track(result, type);
+
+    this.compareSentAndReceived(result, type);
+  }
+
+  parsePeerStats = (res) => {
+    if (!res) {
+      return;
+    }
+    this.compareSentAndReceived(res);
+  }
+
+  startAnalyzer = (peerconnection) => {
+    this.statsInterval = setInterval(() => {
+      this.startStats(peerconnection);
+    }, this.config.interval);
+
+    this.mqaInterval = setInterval(() => {
+      // TODO: get the stats results and send MQA
+    }, this.config.mqaInterval);
+  }
+
+  startStats = (pc) => {
+    const types = ['track', 'transport', 'candidate-pair', 'outbound-rtp', 'outboundrtp', 'inbound-rtp', 'inboundrtp', 'remote-inbound-rtp', 'remote-outbound-rtp'];
+
+    pc.videoTransceiver.sender.getStats().then((res) => {
+      res.forEach((r) => {
+        if (types.includes(r.type)) { this.parseResults(r, 'video'); }
+      });
+    });
+
+    pc.videoTransceiver.receiver.getStats().then((res) => {
+      res.forEach((r) => {
+        if (types.includes(r.type)) { this.parseResults(r, 'video'); }
+      });
+    });
+
+    pc.audioTransceiver.sender.getStats().then((res) => {
+      res.forEach((r) => {
+        if (types.includes(r.type)) { this.parseResults(r, 'audio'); }
+      });
+    });
+
+    pc.audioTransceiver.receiver.getStats().then((res) => {
+      res.forEach((r) => {
+        if (types.includes(r.type)) { this.parseResults(r, 'audio'); }
+      });
+    });
+
+    pc.shareTransceiver.sender.getStats().then((res) => {
+      res.forEach((r) => {
+        if (types.includes(r.type)) { this.parseResults(r, 'share'); }
+      });
+    });
+
+    pc.shareTransceiver.receiver.getStats().then((res) => {
+      res.forEach((r) => {
+        if (types.includes(r.type)) { this.parseResults(r, 'share'); }
+      });
+    });
+  }
+
+  outboundrtp = (result, type) => {
+    if (!result) {
+      return;
+    }
+    if (result.type !== 'outbound-rtp') return;
```

Comment:
Surprised this is passing eslint checks. Needs curly braces and looks like you might be able to combine some of the if statements.